### PR TITLE
Fixes to a couple of issues with the SCH Aetherflow Module

### PR DESCRIPTION
**Pull request by meldontaragon**
+ Drift for aetherflow casts was incorrectly calculated when followed by dissipation
+ Recitation pre-pull usage was counted as an extra aetherflow usage
+ Wasted aetherflow usages weren't calculated for when dissipation was cast while there were remaining aetherflow stacks

### CODE REVIEW

**Review by ayyaruq — CHANGES_REQUESTED** [review]
This nominally looks ok to me but I won't pretend I understand SCH. A few changes for maintainability, but I'd also strongly consider replacing this module with CoreGauge (and writing it in TS) even if that's kinda scope creep. I feel like the gauge module in core can simplify a lot of this by handling accounting such as overcapping for you, so the only real logic becomes the drift detection.

File: src/parser/jobs/sch/modules/Aetherflow.js
Comment (meldontaragon) [reply src/parser/jobs/sch/modules/Aetherflow.js:91]: I'm not personally familiar enough with the project and CoreGauge to implement those changes myself (at least right now) but I can look into it. Otherwise both the suggested edits seem good so I'll throw together a commit and add those to this PR

Comment (ayyaruq) [reply src/parser/jobs/sch/modules/Aetherflow.js]: More of a future thing to do I guess, fae gauge too. WAR has a good example of a fixed value gauge, there's also timer gauges which you can hook events like expiry on. I'm happy to merge this once the structural stuff is changed tho.

**Comment by meldontaragon** [comment]
While adding the suggested changes I noticed a typo in another change I had made. I've corrected that but it's now causing the fake pre-pull Dissipation casts to increment the wasted Aetherflow count so I'll need to write up a fix for that before this should be merged.

**Comment by meldontaragon** [comment]
I realized I forgot to respond again but I implemented the requested changes so this should be good to merge unless there's anything else

**Review by ayyaruq — APPROVED** [review]
Looks ok to me, I still think this module prob needs a proper rewrite but your changes get the job done for now. Can you post a before+after screenshot just to confirm, and then I'll merge?

**Comment by meldontaragon** [comment]
Screenshots of the aetherflow module output for my development build?

**Comment by ayyaruq** [comment]
Yeah, it's prob fine but when I ran your PR I couldn't see any diff, and I'm not familiar enough with SCH to find a log to demonstrate it. If you're sure it works, I'll just yolo merge it.

**Comment by meldontaragon** [comment]
From this log: https://xivanalysis.com/fflogs/DAMczayWBTkHpJKX/8/1

Before and after showing the removal of pre-pull Recitation actions as well the correction of the drift calculation when Dissipation is used (3.64s vs 0.64s)

 Before_1
After_1

Before and after from that same log as above showing the correct Aetherflow usage count without the pre-pull Recitation usage cast being counted

Screenshot_20210130_233616
Screenshot_20210130_233627

From this log: https://xivanalysis.com/fflogs/TDb1AQdkC4pmaFjM/17/7

Before and after with incorrect pre-pull Dissipation being displayed

Screenshot_20210130_233815
Screenshot_20210130_233822

From this log: https://xivanalysis.com/fflogs/17AhBQtqMpbJz2D4/9/1

Before and after showing the stacks lost from an early Dissipation cast

Screenshot_20210130_234207
Screenshot_20210130_234214

**Comment by ayyaruq** [comment]
Awesome, thanks!

**Pull request merged**
